### PR TITLE
implement spike locations

### PR DESCRIPTION
**Pull request by magland**

### CODE REVIEW

File: tests/test_spike_locations.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    ymin = np.min(channel_locations[:, 1])
+    ymax = np.max(channel_locations[:, 1])
+    xspan = xmax - xmin
+    yspan = ymax - ymin
+    if xmax <= xmin:
+        xmin = xmin - 12
+        xmax = xmax + 12
+    if ymax <= ymin:
+        ymin = ymin - 12
+        ymax = ymax + 12
```
Comment (jsoules) [inline tests/test_spike_locations.py:36]: This is weird--why would this happen?

File: tests/test_spike_locations.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    print(url)
+
+def test_spike_locations(*, recording: si.BaseRecording, sorting: si.BaseSorting, hide_unit_selector: bool=False):
+    channel_locations = recording.get_channel_locations().astype(np.float32)
+    xmin = np.min(channel_locations[:, 0])
+    xmax = np.max(channel_locations[:, 0])
+    ymin = np.min(channel_locations[:, 1])
+    ymax = np.max(channel_locations[:, 1])
+    xspan = xmax - xmin
+    yspan = ymax - ymin
```
Comment (jsoules) [inline tests/test_spike_locations.py:30]: This is going to get clobbered by lines 38-39 so you can probably just delete it

File: tests/test_spike_locations.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    for unit_id in sorting.get_unit_ids():
+        spike_times_sec = np.array(sorting.get_unit_spike_train(unit_id=unit_id)) / sorting.get_sampling_frequency()
+        center_x = rng.uniform(xmin, xmax) # fake center for unit
+        center_y = rng.uniform(ymin, ymax) # fake center for unit
+        items.append(
+            vv.SpikeLocationsItem(
+                unit_id=unit_id,
+                spike_times_sec=spike_times_sec.astype(np.float32),
+                x_locations=rng.normal(center_x, 6, spike_times_sec.shape).astype(np.float32), # fake locations
+                y_locations=rng.normal(center_y, 6, spike_times_sec.shape).astype(np.float32)  # fake locations
```
Comment (jsoules) [inline tests/test_spike_locations.py:56]: Is there any particular significance to setting the standard deviation to 6? I'm guessing this is related to spreading the x and y ranges by 24 earlier?

**Review by jsoules — APPROVED** [review]
This all seems reasonable, as far as I can tell. I made a couple notes for clarifications on the `test_spike_locations` script.

What's the overall context for the changes?

File: tests/test_spike_locations.py (hunk 1 above)
Comment (magland) [reply tests/test_spike_locations.py:36]: This happens if all the electrodes have the same x (or y) coordinate

File: tests/test_spike_locations.py (hunk 2 above)
Comment (magland) [reply tests/test_spike_locations.py:30]: good point

File: tests/test_spike_locations.py (hunk 3 above)
Comment (magland) [reply tests/test_spike_locations.py:56]: This is all just a test... was just trying to make the figure look good.

**Pull request merged**